### PR TITLE
Tilted for FCC : Geometry code + Analyzer code

### PR DESCRIPTION
**Pull request by ghugo83**
Tilted for FCC : FULL Geometry code + Analyzer code

The code is very debug style ;p
Warning : The flat barrel modules placement hasn't been touched, the tilted code is 'plugged' on it.
In my opinion there are geometry errors in the code for the flat barrel modules placement.

A test layout cfg files is included in the PR, and provide the following results : 
very_first_tilted_fcc
zoom_outer_tracker_fcc

Results web page is accessible at : 
file:///afs/cern.ch/user/g/ghugo/public/tilted_for_FCC/results/index.html

### CODE REVIEW

**Comment by drasal** [comment]
Hi Gabi, 
thanks a lot, I will have a look tomorrow ...

**Comment by ghugo83** [comment]
- Now all tilted geometry information is displayed on website.
- Created first tilted Inner Tracker cfg file.
- Optimized Inner and Outer Tracker tiltAngles and number of Rods to sth less irrealistic.

**Comment by drasal** [comment]
Gabi, that's great!

**Comment by alkemyst** [comment]
Hi,
  I am testing the code now with the following options:
`tklayout --performance -d -a -N 3000 -n 50000 -T -R -w geometries/CMS_Phase2/OT613_200_IT4025.cfg`
  and I spotted the following changes:
1) The "Material (outer)" tab - in the section "Nuclear interactions" the interactions,  the pixel hits are now taken into account. We can take this as a feature change.
2) *regression* Still in the "Material" tab (both outer and pixel) - in the section "Services details (Full volume)" the material table disappeared and the plot disappeared too :-1:
3) *regression?* in the "Material (total)" section the total material amount changed drastically with a large increase of the component named "Services and Supports under Pixel Tracking Volume" from 0.00015 X₀ to 0.048 X₀ and the component Services and Supports between Pixel and Outer Tracking Volumes from 0.06953 X₀ to 0.15834 X₀
  Looking in the details (full layout material : components) the change is entirely due to "Services under Pixel Tracking Volume" and "Services and supports in interstice"

BEFORE:


AFTER:
fulllayoutmatcategoriestrackingvolume000

**Comment by drasal** [comment]
@alkemyst  Can you, please, revert this PR and keep it still as open? I'm still testing that and have some issues with ... Once I resolve it, we can merge the PR.

**Comment by alkemyst** [comment]
Aye aye, sir

**Comment by alkemyst** [comment]
Done. Reverted with #350 
If you want to make changes and then integrate please
1) merge ghugo83:devLite into your development branch
2) open a new PR from this branch
Thanks

**Comment by drasal** [comment]
I've already done so :-) I'm working on that locally and will merge, once I sort that out...

**Comment by ghugo83** [comment]
Hi,
Hmm as you want but I would not be in favor of an octopus strategy, keeping the branch unmerged in parallell.
Does this bring any regression to the flat placement ?
In my local it gives the exact same results for the flat.
Also for the tilted, things are operational in my branch.
If there is anything to fix, let's do it together this afternoon, and then merge this once and for all :) 
Things can always be improved on the fly afterwards, while i think having a geant unmerged in parallel will cause trouble / extra work.

**Comment by drasal** [comment]
Hi Gabi, 

don't worry, there will be no octopus strategy, your PR will go first to the official devLite.  Just reopen the PR. I will send you several comments ...

**Comment by ghugo83** [comment]
Hi Zbynek,
Oki

**Comment by ghugo83** [comment]
Oki I am changing that on the new PR : https://github.com/tkLayout/tkLayout/pull/351

Sorry I would have liked to stay on the same page, ok whatever ;)

File: source/Analysis/include/AnalyzerGeometry.h [hunk 1]
```diff
@@ -37,6 +37,16 @@ class TProfile;
...
 
+
+// Formatting parameters
+static const int coordPrecision = 3;
+static const int zOverlapPrecision = 5;
+static const int anglePrecision = 1;
+// Some strings for the html formatting
+static const std::string subStart   = "<sub>";      // These only should be needed
+static const std::string subEnd     = "</sub>";
+
```
Comment (drasal) [inline source/Analysis/include/AnalyzerGeometry.h]: Gabi, the global constants are not needed in various analyzers. They should be defined just as local member constants. In addition, subStart, subEnd is already defined in global_constants.h file ... Can you ,please, change it?

**Comment by drasal** [comment]
There are some issues with coding style, but don't worry about it, I will change it by myself. tkLayout-lite is official SW for FCC-hh tracker layout and needs to follow strict coding results. That's why.

File: source/Analysis/include/AnalyzerGeometry.h (hunk 1 above)
Comment (ghugo83) [inline source/Analysis/include/AnalyzerGeometry.h]: Hi Zbynek, yes sure !
Sorry this is just 'first rush' 3000 code lines to make things working :)
The code here is not bound nor even intended at all to be with style, but to provide consistent results :)

Maybe let's look at the geo results themselves at runtime if there are osies, then the style can be improved later (and on dev_gabie as well ;) )

**Comment by drasal** [comment]
Does this version provides the tilted geometry with  run/FCChh_Option3_tilted/FCChh_Option3.cfg? It doesn't for me... Can you, please, check that again? That was actually the reason, why I stopped the PR.

**Comment by ghugo83** [comment]
Hi Zbynek,
Your meeting is in one week, so I would say the style there will always be time a posteriori :p 
No the flat should be on run/FCChh_Option3/FCChh_Option3.cfg 
Let me do the cleaning in cfg files you suggested.
One sec ;)

File: source/Tools/src/GitRevision.cc [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #include <GitRevision.h>
 
 //GitRevision.cc - written by cmake. changes will be lost!
-const std::string GitRevision::revisionNumber = "devLite-798 on git@github.com:drasal/tkLayout.git";
+const std::string GitRevision::revisionNumber = "devLite-900 on git@github.com:ghugo83/tkLayout.git";
```
Comment (drasal) [inline source/Tools/src/GitRevision.cc]: Can you, please, do git checkout  source/Tools/src/GitRevision.cc? GitRevision.cc change is artificial... This is something, what comes out of CMake compilation, if one removes content of build directory without doing make uninstall. That's the weak part of CMake compilation...

File: source/Analysis/include/AnalyzerGeometry.h (hunk 1 above)
Comment (drasal) [inline source/Analysis/include/AnalyzerGeometry.h]: So, for subStart and subEnd, you may directly use definition from global_constants.h :-)

**Comment by drasal** [comment]
Hi Gabi, the meeting is already this week, so I have to postpone my presentation for the next one, anyway...

**Comment by ghugo83** [comment]
Aaaah Zbynek :)
The tilted geometry is in : 
  run/FCChh_Option3_tilted/FCChh_Option3_tilted.cfg
If you run this cfg file evth should work.

Sorry  run/FCChh_Option3_tilted/FCChh_Option3.cfg is a rubbish file which should be deleted

**Comment by ghugo83** [comment]
So to conclude : 
- flat is in  : run/FCChh_Option3/FCChh_Option3.cfg
- tilted is in : run/FCChh_Option3_tilted/FCChh_Option3_tilted.cfg

And run/FCChh_Option3_tilted/FCChh_Option3.cfg is rubbish and should be deleted.
Are you sure you do not have the tilted working with this exact branch on run/FCChh_Option3_tilted/FCChh_Option3_tilted.cfg ?

**Comment by ghugo83** [comment]
PR reopened at : https://github.com/tkLayout/tkLayout/pull/351

**Comment by drasal** [comment]
Sorry, I ran the run/FCChh_Option3_tilted/FCChh_Option3_tilted.cfg, but still didn't work. Let's keep the version as it is, I need to focus on my presentation for the next FCC-hh general meeting now. Unfortunately. We may come back to this PR on Thursday. I will get this PR again and try it out.

**Comment by ghugo83** [comment]
Hi Zbynek,

Ok, I am sorry it doesn't work on yours :( 
It works on my local though, it should not be too much, I will try to investigate and push things if relevant ;)

In any case the FCC layout : file:///afs/cern.ch/user/g/ghugo/public/tilted_for_FCC/results/index.html 
is already not too bad :)

Frustrating! Let me look :)

**Comment by ghugo83** [comment]
Sorry this is a mess! For some reason not possible to reopen this PR :/
The continuation is at https://github.com/tkLayout/tkLayout/pull/352 
I am pushing improvements there.

I did not find any issue, are you sure you tried with the exact ghugo83/tkLayout/devLite ?

**Comment by ghugo83** [comment]
Sorry guys, becoming crazy with all the merges.
Basically I think we have a merge issue.
The first 'revert' also reverted my dev into my github branch. 
Each time, trying to push again a dev for the tilted into tkLayout/devLite was not working, even with the correct changes in a correct local branch.
Let me try an experiment ;p

**Pull request merged**